spec/API_specification/linear_algebra_functions.md

```diff
@@ -265,9 +265,30 @@ TODO
 TODO
 
 (function-slogdet)=
-### slogdet()
+### slogdet(x, /)
 
-TODO
+Returns the sign and the natural logarithm of the absolute value of the determinant of a square matrix (or stack of square matrices) `x`.
+
+```{note}
+
+The purpose of this function is to calculate the determinant more accurately when the determinant is either very small or very large, as calling `det` may overflow or underflow.
+```
+
+#### Parameters
+
+-   **x**: _&lt;array&gt;_
+
+    -   input array having shape `(..., M, M)` and whose innermost two dimensions form square matrices. Must have a data type of either `float32` or `float64`.
```

Comment:
Why the limitation on input dtype? I checked numpy and cupy just now, they're perfectly happy with integer input. Output is always floating point.

Reply:
This currently follows [`det`](https://github.com/data-apis/array-api/blob/56fb2fbdec1f8d31fdb0ac92cf8540aa6c347222/spec/API_specification/linear_algebra_functions.md#detx-). In general, we've limited linalg specs to floating-point data types for consistency (i.e., not having a patchwork of linalg APIs supporting different data types). Can relax here (and for `det`), if desired.

Reply:
If this is a concern (disclaimer: I don't really care), perhaps we could say the input would be promoted to floating-point types following the type promotion rule?

Reply:
As the current type promotion rules do not address mixed-kind promotion, supporting integer input would be ill-defined. The only behavior the spec can presently define is when provided floating-point input.

Reply:
Loosened the dtype requirements to "should have a floating-point data type". This recommendation stems from needing to cast to a floating-point data type and mixed-kind promotions being undefined in the spec.